<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>830</width>
    <height>551</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Dreamer</string>
  </property>
  <property name="windowIcon">
   <iconset resource="dreamer.qrc">
    <normaloff>:/icons/icons/Logo.png</normaloff>:/icons/icons/Logo.png</iconset>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QHBoxLayout" name="horizontalLayout">
    <property name="leftMargin">
     <number>0</number>
    </property>
    <property name="topMargin">
     <number>0</number>
    </property>
    <property name="rightMargin">
     <number>0</number>
    </property>
    <property name="bottomMargin">
     <number>0</number>
    </property>
    <item>
     <widget class="QSplitter" name="splitter">
      <property name="orientation">
       <enum>Qt::Horizontal</enum>
      </property>
      <widget class="QWidget" name="generator_widget" native="true">
       <layout class="QVBoxLayout" name="verticalLayout">
        <item>
         <widget class="QComboBox" name="mode">
          <property name="enabled">
           <bool>true</bool>
          </property>
          <item>
           <property name="text">
            <string>Text to Image</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>Image to Image</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>InPainting</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>Upscaling</string>
           </property>
          </item>
         </widget>
        </item>
        <item>
         <widget class="QPlainTextEdit" name="prompt_edit"/>
        </item>
        <item>
         <widget class="QGroupBox" name="prompt_settings">
          <property name="title">
           <string>Prompt Settings</string>
          </property>
          <layout class="QVBoxLayout" name="verticalLayout_4">
           <item>
            <layout class="QHBoxLayout" name="horizontalLayout_2">
             <property name="sizeConstraint">
              <enum>QLayout::SetMinimumSize</enum>
             </property>
             <item>
              <widget class="QLabel" name="label_2">
               <property name="sizePolicy">
                <sizepolicy hsizetype="Fixed" vsizetype="Preferred">
                 <horstretch>0</horstretch>
                 <verstretch>0</verstretch>
                </sizepolicy>
               </property>
               <property name="minimumSize">
                <size>
                 <width>70</width>
                 <height>0</height>
                </size>
               </property>
               <property name="text">
                <string>Size</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QSpinBox" name="image_width">
               <property name="toolTip">
                <string>Image width</string>
               </property>
               <property name="minimum">
                <number>64</number>
               </property>
               <property name="maximum">
                <number>2048</number>
               </property>
               <property name="singleStep">
                <number>64</number>
               </property>
               <property name="value">
                <number>512</number>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QLabel" name="label_3">
               <property name="maximumSize">
                <size>
                 <width>10</width>
                 <height>16777215</height>
                </size>
               </property>
               <property name="text">
                <string>x</string>
               </property>
               <property name="alignment">
                <set>Qt::AlignCenter</set>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QSpinBox" name="image_height">
               <property name="toolTip">
                <string>Image height</string>
               </property>
               <property name="minimum">
                <number>64</number>
               </property>
               <property name="maximum">
                <number>2048</number>
               </property>
               <property name="singleStep">
                <number>64</number>
               </property>
               <property name="value">
                <number>512</number>
               </property>
              </widget>
             </item>
            </layout>
           </item>
          </layout>
         </widget>
        </item>
        <item>
         <widget class="QGroupBox" name="image_settings">
          <property name="visible">
           <bool>false</bool>
          </property>
          <property name="title">
           <string>Image Settings</string>
          </property>
          <layout class="QVBoxLayout" name="verticalLayout_3">
           <item>
            <layout class="QHBoxLayout" name="horizontalLayout_7">
             <item>
              <widget class="QLabel" name="label_8">
               <property name="sizePolicy">
                <sizepolicy hsizetype="Fixed" vsizetype="Preferred">
                 <horstretch>0</horstretch>
                 <verstretch>0</verstretch>
                </sizepolicy>
               </property>
               <property name="minimumSize">
                <size>
                 <width>70</width>
                 <height>0</height>
                </size>
               </property>
               <property name="text">
                <string>Preserve</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QDoubleSpinBox" name="preserve_image">
               <property name="toolTip">
                <string>Prompt influence to image, (0-0.99)
0: Keep the image
0,99: Will affect hugely on the image</string>
               </property>
               <property name="minimum">
                <double>0.010000000000000</double>
               </property>
               <property name="maximum">
                <double>1.000000000000000</double>
               </property>
               <property name="singleStep">
                <double>0.010000000000000</double>
               </property>
               <property name="value">
                <double>0.250000000000000</double>
               </property>
              </widget>
             </item>
            </layout>
           </item>
          </layout>
         </widget>
        </item>
        <item>
         <widget class="QGroupBox" name="general_settings">
          <property name="title">
           <string>General Settings</string>
          </property>
          <property name="checkable">
           <bool>false</bool>
          </property>
          <layout class="QVBoxLayout" name="verticalLayout_2">
           <item>
            <layout class="QHBoxLayout" name="horizontalLayout_3">
             <item>
              <widget class="QLabel" name="label_4">
               <property name="sizePolicy">
                <sizepolicy hsizetype="Fixed" vsizetype="Preferred">
                 <horstretch>0</horstretch>
                 <verstretch>0</verstretch>
                </sizepolicy>
               </property>
               <property name="minimumSize">
                <size>
                 <width>70</width>
                 <height>0</height>
                </size>
               </property>
               <property name="text">
                <string>Seed</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QLineEdit" name="seed">
               <property name="enabled">
                <bool>false</bool>
               </property>
               <property name="minimumSize">
                <size>
                 <width>75</width>
                 <height>0</height>
                </size>
               </property>
               <property name="toolTip">
                <string>Generator seed</string>
               </property>
               <property name="text">
                <string>0</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QCheckBox" name="random_seed">
               <property name="enabled">
                <bool>true</bool>
               </property>
               <property name="text">
                <string>Random</string>
               </property>
               <property name="checked">
                <bool>true</bool>
               </property>
              </widget>
             </item>
            </layout>
           </item>
           <item>
            <layout class="QHBoxLayout" name="horizontalLayout_11">
             <item>
              <widget class="QLabel" name="label_11">
               <property name="minimumSize">
                <size>
                 <width>70</width>
                 <height>0</height>
                </size>
               </property>
               <property name="maximumSize">
                <size>
                 <width>70</width>
                 <height>16777215</height>
                </size>
               </property>
               <property name="text">
                <string>Variation</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QDoubleSpinBox" name="variation">
               <property name="enabled">
                <bool>false</bool>
               </property>
               <property name="toolTip">
                <string>Seed variation
0: Same seed will have same result
1: Same seed will produce different variation</string>
               </property>
               <property name="maximum">
                <double>1.000000000000000</double>
               </property>
               <property name="singleStep">
                <double>0.100000000000000</double>
               </property>
              </widget>
             </item>
            </layout>
           </item>
           <item>
            <layout class="QHBoxLayout" name="horizontalLayout_4">
             <item>
              <widget class="QLabel" name="label_5">
               <property name="sizePolicy">
                <sizepolicy hsizetype="Fixed" vsizetype="Preferred">
                 <horstretch>0</horstretch>
                 <verstretch>0</verstretch>
                </sizepolicy>
               </property>
               <property name="minimumSize">
                <size>
                 <width>70</width>
                 <height>0</height>
                </size>
               </property>
               <property name="text">
                <string>Steps</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QSlider" name="steps">
               <property name="toolTip">
                <string>The steps required to generate image
Higher means more detail will be produced but will take more time</string>
               </property>
               <property name="minimum">
                <number>1</number>
               </property>
               <property name="maximum">
                <number>200</number>
               </property>
               <property name="value">
                <number>25</number>
               </property>
               <property name="orientation">
                <enum>Qt::Horizontal</enum>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QLabel" name="steps_value">
               <property name="text">
                <string>025</string>
               </property>
              </widget>
             </item>
            </layout>
           </item>
           <item>
            <layout class="QHBoxLayout" name="horizontalLayout_5">
             <item>
              <widget class="QLabel" name="label_6">
               <property name="sizePolicy">
                <sizepolicy hsizetype="Fixed" vsizetype="Preferred">
                 <horstretch>0</horstretch>
                 <verstretch>0</verstretch>
                </sizepolicy>
               </property>
               <property name="minimumSize">
                <size>
                 <width>70</width>
                 <height>0</height>
                </size>
               </property>
               <property name="text">
                <string>Images</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QSlider" name="iteration">
               <property name="toolTip">
                <string>Number of image to be generated</string>
               </property>
               <property name="minimum">
                <number>1</number>
               </property>
               <property name="orientation">
                <enum>Qt::Horizontal</enum>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QLabel" name="iteration_value">
               <property name="text">
                <string>001</string>
               </property>
              </widget>
             </item>
            </layout>
           </item>
           <item>
            <layout class="QHBoxLayout" name="horizontalLayout_6">
             <item>
              <widget class="QLabel" name="label_7">
               <property name="sizePolicy">
                <sizepolicy hsizetype="Fixed" vsizetype="Preferred">
                 <horstretch>0</horstretch>
                 <verstretch>0</verstretch>
                </sizepolicy>
               </property>
               <property name="minimumSize">
                <size>
                 <width>70</width>
                 <height>0</height>
                </size>
               </property>
               <property name="text">
                <string>Sampler</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QComboBox" name="sampler">
               <property name="toolTip">
                <string>Image sampler type</string>
               </property>
               <property name="currentIndex">
                <number>7</number>
               </property>
               <item>
                <property name="text">
                 <string>plms</string>
                </property>
               </item>
               <item>
                <property name="text">
                 <string>ddim</string>
                </property>
               </item>
               <item>
                <property name="text">
                 <string>k_dpm_2_a</string>
                </property>
               </item>
               <item>
                <property name="text">
                 <string>k_dpm_2</string>
                </property>
               </item>
               <item>
                <property name="text">
                 <string>k_euler_a</string>
                </property>
               </item>
               <item>
                <property name="text">
                 <string>k_euler</string>
                </property>
               </item>
               <item>
                <property name="text">
                 <string>k_heun</string>
                </property>
               </item>
               <item>
                <property name="text">
                 <string>k_lms</string>
                </property>
               </item>
              </widget>
             </item>
            </layout>
           </item>
           <item>
            <layout class="QHBoxLayout" name="horizontalLayout_12">
             <item>
              <widget class="QLabel" name="label_12">
               <property name="minimumSize">
                <size>
                 <width>70</width>
                 <height>0</height>
                </size>
               </property>
               <property name="maximumSize">
                <size>
                 <width>70</width>
                 <height>16777215</height>
                </size>
               </property>
               <property name="text">
                <string>Strength</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QDoubleSpinBox" name="strength">
               <property name="toolTip">
                <string>How strongly the prompt influences the image</string>
               </property>
               <property name="decimals">
                <number>1</number>
               </property>
               <property name="minimum">
                <double>1.000000000000000</double>
               </property>
               <property name="maximum">
                <double>20.000000000000000</double>
               </property>
               <property name="singleStep">
                <double>0.500000000000000</double>
               </property>
               <property name="value">
                <double>7.500000000000000</double>
               </property>
              </widget>
             </item>
            </layout>
           </item>
           <item>
            <layout class="QHBoxLayout" name="horizontalLayout_10">
             <item>
              <widget class="QLabel" name="label_9">
               <property name="minimumSize">
                <size>
                 <width>70</width>
                 <height>0</height>
                </size>
               </property>
               <property name="maximumSize">
                <size>
                 <width>70</width>
                 <height>16777215</height>
                </size>
               </property>
               <property name="text">
                <string>Upscale</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QSpinBox" name="upscale">
               <property name="toolTip">
                <string>Upscale resolution</string>
               </property>
               <property name="maximum">
                <number>10</number>
               </property>
               <property name="value">
                <number>1</number>
               </property>
              </widget>
             </item>
            </layout>
           </item>
           <item>
            <layout class="QHBoxLayout" name="horizontalLayout_8">
             <item>
              <widget class="QLabel" name="label_10">
               <property name="minimumSize">
                <size>
                 <width>70</width>
                 <height>0</height>
                </size>
               </property>
               <property name="maximumSize">
                <size>
                 <width>70</width>
                 <height>16777215</height>
                </size>
               </property>
               <property name="text">
                <string>Reconstruct</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QSpinBox" name="sharpness">
               <property name="toolTip">
                <string>Reconstruct Face using ESRGAN</string>
               </property>
               <property name="value">
                <number>0</number>
               </property>
              </widget>
             </item>
            </layout>
           </item>
           <item>
            <widget class="QCheckBox" name="seamless">
             <property name="text">
              <string>Seamless / Tileable</string>
             </property>
            </widget>
           </item>
          </layout>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="generate_btn">
          <property name="text">
           <string>Generate</string>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
      <widget class="QMdiArea" name="image_area">
       <property name="sizePolicy">
        <sizepolicy hsizetype="Minimum" vsizetype="Minimum">
         <horstretch>0</horstretch>
         <verstretch>0</verstretch>
        </sizepolicy>
       </property>
       <property name="minimumSize">
        <size>
         <width>512</width>
         <height>512</height>
        </size>
       </property>
       <property name="background">
        <brush brushstyle="SolidPattern">
         <color alpha="255">
          <red>128</red>
          <green>128</green>
          <blue>128</blue>
         </color>
        </brush>
       </property>
      </widget>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>830</width>
     <height>21</height>
    </rect>
   </property>
  </widget>
  <widget class="QToolBar" name="toolBar">
   <property name="windowTitle">
    <string>toolBar_2</string>
   </property>
   <property name="orientation">
    <enum>Qt::Vertical</enum>
   </property>
   <attribute name="toolBarArea">
    <enum>LeftToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
   <addaction name="actionNew"/>
   <addaction name="actionOpenImage"/>
   <addaction name="actionSaveImage"/>
   <addaction name="separator"/>
   <addaction name="actionInPainting"/>
   <addaction name="actionBrushColor"/>
   <addaction name="actionBrushSize"/>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
  <action name="actionNew">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="checked">
    <bool>true</bool>
   </property>
   <property name="icon">
    <iconset resource="dreamer.qrc">
     <normaloff>:/icons/icons/idea.png</normaloff>:/icons/icons/idea.png</iconset>
   </property>
   <property name="text">
    <string>New</string>
   </property>
   <property name="toolTip">
    <string>Generate New Image
Shortcut : Ctrl+N</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+N</string>
   </property>
  </action>
  <action name="actionInPainting">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="icon">
    <iconset resource="dreamer.qrc">
     <normaloff>:/icons/icons/brush.png</normaloff>:/icons/icons/brush.png</iconset>
   </property>
   <property name="text">
    <string>InPainting</string>
   </property>
   <property name="toolTip">
    <string>Paint Mode
Shortcut : Ctrl+P</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+P</string>
   </property>
  </action>
  <action name="actionSaveImage">
   <property name="icon">
    <iconset resource="dreamer.qrc">
     <normaloff>:/icons/icons/save.png</normaloff>:/icons/icons/save.png</iconset>
   </property>
   <property name="text">
    <string>Save Image</string>
   </property>
   <property name="toolTip">
    <string>Save Image
Shortcut : Ctrl+S</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+S</string>
   </property>
  </action>
  <action name="actionBrushColor">
   <property name="text">
    <string>Color</string>
   </property>
   <property name="toolTip">
    <string>Change Brush ColorShortcut : Ctrl:I</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+I</string>
   </property>
  </action>
  <action name="actionBrushSize">
   <property name="icon">
    <iconset resource="dreamer.qrc">
     <normaloff>:/icons/icons/stroke.png</normaloff>:/icons/icons/stroke.png</iconset>
   </property>
   <property name="text">
    <string>Size</string>
   </property>
   <property name="toolTip">
    <string>Change Brush Size
Shortcut : Ctrl+B</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+B</string>
   </property>
  </action>
  <action name="actionOpenImage">
   <property name="icon">
    <iconset resource="dreamer.qrc">
     <normaloff>:/icons/icons/open.png</normaloff>:/icons/icons/open.png</iconset>
   </property>
   <property name="text">
    <string>Open Image</string>
   </property>
   <property name="toolTip">
    <string>Open Image
Shortcut : Ctrl+O</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+O</string>
   </property>
  </action>
 </widget>
 <resources>
  <include location="dreamer.qrc"/>
 </resources>
 <connections/>
</ui>
